client/post-editor/editor-gutenberg-blocks-warning-dialog/index.jsx

```diff
@@ -0,0 +1,143 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { identity, noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getEditorRawContent, getEditorPostId } from 'state/ui/editor/selectors';
+import Dialog from 'components/dialog';
+import { setSelectedEditor } from 'state/selected-editor/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
+import getGutenbergEditorUrl from 'state/selectors/get-gutenberg-editor-url';
+import { openPostRevisionsDialog } from 'state/posts/revisions/actions';
+
+class EditorGutenbergBlocksWarningDialog extends Component {
+	static propTypes = {
+		translate: PropTypes.func,
+		postContent: PropTypes.string,
+		siteId: PropTypes.number,
+		gutenbergUrl: PropTypes.string,
+		switchToGutenberg: PropTypes.func,
+		openPostRevisionsDialog: PropTypes.func,
+	};
+
+	static defaultProps = {
+		translate: identity,
+		postContent: null,
+		siteId: null,
+		gutenbergUrl: null,
+		switchToGutenberg: noop,
+		openPostRevisionsDialog: noop,
+	};
+
+	state = {
+		isDialogVisible: false,
+		forceClassic: false,
+	};
+
+	shouldComponentUpdate( nextProps, nextState ) {
+		return this.state.isDialogVisible !== nextState.isDialogVisible;
+	}
+
+	static getDerivedStateFromProps( props, state ) {
+		const { postContent } = props;
+		const { forceClassic } = state;
+
+		const hasGutenbergBlocks = content => !! content && content.indexOf( '<!-- wp:' ) !== -1;
+
+		const isDialogVisible = ! forceClassic && hasGutenbergBlocks( postContent );
+
+		return {
+			isDialogVisible,
+		};
+	}
+
+	onUseClassic = () => {
+		this.setState( {
+			forceClassic: true,
+		} );
+	};
```

Comment:
we typically avoid using `onXYZ` names and prefer instead names that tersely describe what they do, like `useClassicEditor` and `switchToGutenberg`

[coding guidelines](https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/javascript.md) check out the section `React Components`

Reply:
Gotcha! I also like that convention, but I don't know why I was using the React convention for props names. 

Changed in 6ce013b.